Migration notes about `<uses-sdk/>` element (#1624)
dotnet/docs-maui: docs/migration/android-projects.md

> [!NOTE]
> Referencing an Android Wear project from an Android app isn't supported.

> ## Changes to `AndroidManifest.xml`

Reviewer: ```suggestion
## Changes to AndroidManifest.xml
```

Reviewer: Code formatting isn't applied to headings. Alternatively this could be "Changes to the Android manifest".

In Xamarin.Android, Java, and Kotlin Android projects, the `<uses-sdk/>` element denotes the minimum Android version your app supports, as well as the target Android version your app is compiled against:

```xml
﻿<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" 
    android:versionCode="1" 
    android:versionName="1.0" 
    package="com.companyname.myapp">
  <uses-sdk android:minSdkVersion="21" android:targetSdkVersion="33" />
  <application android:icon="@mipmap/ic_launcher" android:label="@string/app_name" android:theme="@style/AppTheme" />
</manifest>
```

> See the [Android documentation][uses-sdk] for more information about the `<uses-sdk/>` element.

Reviewer: ```suggestion
See the [Android documentation](https://developer.android.com/guide/topics/manifest/uses-sdk-element) for more information about the `<uses-sdk/>` element.
```

> In .NET 6+ Android applications, there are MSBuild properties to set these values. Using the MSBuild properties has other benefits. In most cases the `<uses-sdk/>` element should be removed in favor of values in your project's `.csproj` file:

Reviewer: ```suggestion
In .NET 6+ Android apps, there are MSBuild properties to set these values. Using the MSBuild properties has other benefits. In most cases the `<uses-sdk/>` element should be removed in favor of values in your project's `.csproj` file:
```

```xml
<Project>
  <PropertyGroup>
    <TargetFramework>net6.0-android</TargetFramework>
    <SupportedOSPlatformVersion>21</SupportedOSPlatformVersion>
  </PropertyGroup>
</Project>
```

> In this example, `net6.0-android` is shorthand for `net6.0-android31.0`. While in .NET 7, `net7.0-android` is shorthand for `net7.0-android33.0`. Future versions of .NET will track the latest Android version available at the time of the .NET release.

Reviewer: Wondering if the .NET 6 examples are required, what with it being out of support for MAUI?

Author: The issue is `net7.0-android33` is the only option, and we probably won't have `net7.0-android34`. .NET 6 happens to have better examples, but I'm open...

> `TargetFramework` maps to `android:targetSdkVersion`, at build time this value will automatically be included in the `<uses-sdk/>` element for you. The benefit of using `TargetFramework` in this way, is we can give you the matching C# binding for Android API 31 for `net6.0-android31.0`. Android releases independently of the .NET release cycle, so we have the flexibility to opt into `net6.0-android32.0` when a binding is available sometime after .NET 6's release.

Reviewer: ```suggestion
`TargetFramework` maps to `android:targetSdkVersion`. At build time this value will automatically be included in the `<uses-sdk/>` element for you. The benefit of using `TargetFramework` in this way, is you are given the matching C# binding for Android API 31 for `net6.0-android31.0`. Android releases independently of the .NET release cycle, so we have the flexibility to opt into `net6.0-android32.0` when a binding is available sometime after .NET 6's release.
```

> Likewise `SupportedOSPlatformVersion` maps to `android:minSdkVersion`, at build time this value will automatically be included in the `<uses-sdk/>` element for you. Android APIs are decorated with the [`SupportedOSPlatformAttribute`][supportedattr] so that you will get build warnings for calling APIs that are only available for some of the Android versions your application can run on:

Reviewer: ```suggestion
Similarly, `SupportedOSPlatformVersion` maps to `android:minSdkVersion`. At build time this value will automatically be included in the `<uses-sdk/>` element for you. Android APIs are decorated with the <xref:System.Runtime.Versioning.SupportedOSPlatformAttribute> so that you will get build warnings for calling APIs that are only available for some of the Android versions your app can run on:
```

```
error CA1416: This call site is reachable on 'Android' 21.0 and later. `ConnectivityManager.ActiveNetwork` is only supported on: 'Android' 23.0 and later.
```

> To safely use this API, you can declare a higher `SupportedOSPlatformVersion` in your project or use the [`OperatingSystem`][operatingsystem] API at runtime:

Reviewer: ```suggestion
To safely use this API, you can declare a higher `SupportedOSPlatformVersion` in your project or use the <xref:System.OperatingSystem.IsAndroidVersionAtLeast%2A> API at runtime:
```

```csharp
if (OperatingSystem.IsAndroidVersionAtLeast(23))
{
    // Use the API here
}
```

> [uses-sdk]: https://developer.android.com/guide/topics/manifest/uses-sdk-element

Reviewer: Delete.

> [supportedattr]: https://learn.microsoft.com/dotnet/api/system.runtime.versioning.supportedosplatformattribute

Reviewer: Delete

> [operatingsystem]: https://learn.microsoft.com/dotnet/api/system.operatingsystem.isandroidversionatleast

Reviewer: Delete.
